RFCs/005-DynamicData-Attribute.md

```diff
@@ -0,0 +1,51 @@
+# RFC 005- DynamicData Attribute for Data Driven Tests
+
+## Summary
+This details the MSTest V2 framework attribute "DynamicData" for data driven tests where data can be specified as property.
+
+## Motivation
+Often times, the inline test data need to be shared between tests. There could be more than one test cases in different files which will we using same test data. In most of these situations, user end up declaring same inline data for multiple test cases using "DataRow" attribute. Test framework should provides an attribute so that test data, declared as property, can be
+* Easily re-used
+* Easily organized
+
+## Detailed Design
+
+### Requirements
+1. Test data declared as property should be easily asscoaited with test case.
+2. Test data declared as property should be associated with more than one test cases, if required.
+
+### Proposed solution
+Here is a solution that meets the above requirements:
+
+The property storing test data should be declared as static.
+```
+    static object[] ReusableTestData =
```

Comment:
Would that force users to type more? I'd prefer to type even less (not repeat `object[]` for every row) by using 2d array:

    static object[,] ReusableTestData = new object[,]
    {
        {1,2,3},
        {4,5,6}
    };

DynamicData could support multiple input types. Ideas: `IEnumerable<object[]>`, `ICollection<object[]>`, `object[][]`, `object[,]`, `Action<...>`.